homeassistant/components/motioneye/__init__.py

```diff
@@ -0,0 +1,263 @@
+                CONF_CONFIG_ENTRY: config_entry,
+            },
+            data=config_entry.data,
+        )
+    )
+
+
+async def _add_camera(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    client: MotionEyeClient,
+    entry: ConfigEntry,
+    camera_id: int,
+    camera: dict[str, Any],
+    device_id: str,
+) -> None:
+    """Add a motionEye camera to hass."""
+
+    device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, device_id)},
+        manufacturer=MOTIONEYE_MANUFACTURER,
+        model=MOTIONEYE_MANUFACTURER,
+        name=camera[KEY_NAME],
+    )
+
+    async_dispatcher_send(
+        hass,
+        SIGNAL_CAMERA_ADD.format(entry.entry_id),
+        camera,
+    )
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up motionEye from a config entry."""
+    client = create_motioneye_client(
+        entry.data[CONF_URL],
+        admin_username=entry.data.get(CONF_ADMIN_USERNAME),
+        admin_password=entry.data.get(CONF_ADMIN_PASSWORD),
+        surveillance_username=entry.data.get(CONF_SURVEILLANCE_USERNAME),
+        surveillance_password=entry.data.get(CONF_SURVEILLANCE_PASSWORD),
+    )
+
+    try:
+        await client.async_client_login()
+    except MotionEyeClientInvalidAuthError:
+        await client.async_client_close()
+        await _create_reauth_flow(hass, entry)
+        return False
+    except MotionEyeClientError as exc:
+        await client.async_client_close()
+        raise ConfigEntryNotReady from exc
+
+    async def async_update_data() -> dict[str, Any] | None:
+        try:
+            return await client.async_get_cameras()
+        except MotionEyeClientError as exc:
+            raise UpdateFailed("Error communicating with API") from exc
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=DOMAIN,
+        update_method=async_update_data,
+        update_interval=DEFAULT_SCAN_INTERVAL,
+    )
+    hass.data[DOMAIN][entry.entry_id] = {
+        CONF_CLIENT: client,
+        CONF_COORDINATOR: coordinator,
+        CONF_ON_UNLOAD: [],
+    }
+
+    current_cameras: set[str] = set()
+    device_registry = await dr.async_get_registry(hass)
+
+    def _async_process_motioneye_cameras() -> None:
+        """Process motionEye camera additions and removals."""
+        inbound_camera: set[str] = set()
+        if KEY_CAMERAS not in coordinator.data:
+            return
+
+        for camera in coordinator.data[KEY_CAMERAS]:
+            if not is_acceptable_camera(camera):
+                return
+            camera_id = camera[KEY_ID]
+            device_unique_id = get_motioneye_device_unique_id(entry.entry_id, camera_id)
+            inbound_camera.add(device_unique_id)
+
+            if device_unique_id in current_cameras:
+                continue
+            current_cameras.add(device_unique_id)
+            hass.async_create_task(
```

Comment:
Don't create a task for it. Just inline it here. All actions are callbacks, so it's faster to not create tasks.